v2/internal/controllers/samples_test.go

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/emirpasic/gods/maps/linkedhashmap"
```

Comment:
I'm not sure that I follow why we need `linkedhashmap`. I get the impression it's for:
> // Using recursion here to maintain the order of creation and deletion of resources

but I think if you (as discussed down below) just simplify the deletion case into deleting the RG at the end, that you can also simplify the creation case by just applying everything in whatever order you want. Apply order shouldn't matter because the reconciler should work it all out. Then you can just drop the usage of `linkedhashmap` and simplify your `createAndDeleteResourceTree` function some?

Were there specific cases you remember where the creation order really mattered? AFAIK it shouldn't matter for anything.

Reply:
We're testing each by using `CreateResourceAndWait`, So if we apply everything in whatever order, and if owner does not already exist for that resource, the test won't proceed. Linkedhashmap is used here to make sure the resources and refs are created in order made sure everything gets created successfully.

Reply:
Hmmm, wondering if we should just split the resources into alpha/beta and then `CreateResourcesAndWait` the whole alpha/beta set? AFAIK order shouldn't matter, although maybe it does for secrets right now as reported in #2385

Reply:
Also, we can't be using random name for all the tests, as we've got armIDs, refs and links within the resource or child resources which should match the name of resource.